stock_available_to_promise_release/models/stock_move.py

```diff
@@ -567,14 +604,17 @@ def _split_origins(self, origins):
                 new_origin_moves |= self.create(new_move_vals)
                 break
             origins -= origin
+        # And then do the reservation again
+        origins._action_assign()
+        new_origin_moves._action_assign()
         return new_origin_moves
 
     def _search_picking_for_assignation_domain(self):
         domain = super()._search_picking_for_assignation_domain()
         if self.env.context.get("release_available_to_promise"):
             force_new_picking = not self.rule_id.no_backorder_at_release
             if force_new_picking:
-                domain = expression.AND([domain, [("id", "!=", self.picking_id.id)]])
+                domain = expression.AND([domain, [("id", ">", self.picking_id.id)]])
```

Comment:
A comment to explain why you use '>' here would be welcome.

Reply:
done